editor.planx.uk/src/pages/FlowEditor/components/Team/tests/TeamMembers.addNewEditor.test.tsx

```diff
@@ -1,33 +1,85 @@
+jest.mock("pages/FlowEditor/lib/store/team.ts", () => ({
+  teamStore: jest.fn().mockImplementation(() => ({
+    fetchCurrentTeam: jest.fn().mockResolvedValue({
+      team_id: 1,
+      role: "teamEditor",
+      user_id: 1,
+      __typename: "team_members",
+    }),
+  })),
+}));
+jest.mock("pages/FlowEditor/components/Team/queries/createUser.tsx", () => ({
+  createUser: jest.fn().mockResolvedValue({
+    id: 1,
+    email: "mickeymouse@email.com",
+    first_name: "Mickey",
+    last_name: "Mouse",
+    is_platform_admin: false,
+    __typename: "users",
+  }),
+}));
+jest.mock("pages/FlowEditor/components/Team/queries/addUserToTeam.tsx", () => ({
+  addUserToTeam: jest.fn().mockResolvedValue({
+    team_id: 1,
+    role: "teamEditor",
+    user_id: 1,
+    __typename: "team_members",
+  }),
+}));
 
-describe("when a user views the Team members screen with the ADD_NEW_EDITOR feature flag enabled", () => {
+describe("when a user with the ADD_NEW_EDITOR feature flag enabled presses 'add a new editor'", () => {
   beforeEach(async () => {
-    await setupTeamMembersScreen();
-  });
-  it("shows the 'add new editor' button", async () => {
+    const user = await setupTeamMembersScreen();
     const teamEditorsTable = screen.getByTestId("team-editors");
-    await within(teamEditorsTable).findByText("Add a new editor");
+    const addEditorButton = await within(teamEditorsTable).findByText(
+      "Add a new editor",
+    );
+    user.click(addEditorButton);
+  });
+  it("opens the modal and displays the input fields", async () => {
+    expect(await screen.findByTestId("modal-create-user-button")).toBeVisible();
+    expect(await screen.findByLabelText("First name")).toBeVisible();
   });
 });
 
-describe("when a user with the ADD_NEW_EDITOR feature flag enabled presses 'add a new editor'", () => {
+describe("when a user fills in the 'add a new editor' form correctly", () => {
   beforeEach(async () => {
     const user = await setupTeamMembersScreen();
     const teamEditorsTable = screen.getByTestId("team-editors");
     const addEditorButton = await within(teamEditorsTable).findByText(
-      "Add a new editor"
+      "Add a new editor",
     );
     user.click(addEditorButton);
+    const addNewEditorModal = await screen.findByTestId("modal-create-user");
+    await userEntersInput("First name", "Mickey", addNewEditorModal);
+    await userEntersInput("Last name", "Mouse", addNewEditorModal);
+    await userEntersInput(
+      "Email address",
+      "mickeymouse@email.com",
+      addNewEditorModal,
+    );
+
+    const createUserButton = await screen.findByTestId(
+      "modal-create-user-button",
+    );
+    user.click(createUserButton);
   });
-  it("opens the modal and displays the input fields", async () => {
-    expect(await screen.findByTestId("modal-create-user-button")).toBeVisible();
-    expect(await screen.findByLabelText("First name")).toBeVisible();
+
+  it("adds the new user row to the Team Editors table", async () => {
```

Comment:
I wanted an `it` block for 'it closes the modal` but had to give up - it was repeatedly throwing false positives (passing even though I don't believe the modal was closed). Would welcome any thoughts on whether I should continue trying to get a test in for that.

Reply:
I managed to fix this, I think I was just missing the right combination of `queryBy` and `waitFor` but it turns out that react testing library has a [good article on testing for disappearance](https://testing-library.com/docs/guide-disappearance). This is in the follow-on PR: https://github.com/theopensystemslab/planx-new/pull/3543